segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AbstractConnectionListener.java

```diff
@@ -72,14 +81,45 @@
     @Getter
     private final boolean enableTlsReload; // whether to reload TLS certificate when the certificate changes
 
+    private final String type;
     private final String pathToTlsCertFile;
     private final String pathToTlsKeyFile;
     private final String[] tlsProtocolVersion;
 
     private FileModificationMonitor tlsCertFileModificationMonitor; // used only if tls reload is enabled
 
+    /**
+     * Creates a new instance of the AdminConnectionListener class.
+     *
+     * @param enableTls          Whether to enable SSL/TLS.
+     * @param enableTlsReload    Whether to reload TLS when the X.509 certificate file is replaced.
+     * @param host               The name of the host to listen to.
+     * @param port               The port to listen on.
+     * @param certFile           Path to the certificate file to be used for TLS.
+     * @param keyFile            Path to be key file to be used for TLS.
+     * @param tlsProtocolVersion the version of the TLS protocol
+     */
     public AbstractConnectionListener(boolean enableTls, boolean enableTlsReload, String host, int port,
                                       String certFile, String keyFile, String[] tlsProtocolVersion) {
+        this(enableTls, enableTlsReload, host, port, certFile, keyFile, tlsProtocolVersion, null, null);
+    }
+
+    /**
+     * Creates a new instance of the AdminConnectionListener class with HealthServiceManager.
+     *
+     * @param enableTls          Whether to enable SSL/TLS.
+     * @param enableTlsReload    Whether to reload TLS when the X.509 certificate file is replaced.
+     * @param host               The name of the host to listen to.
+     * @param port               The port to listen on.
+     * @param certFile           Path to the certificate file to be used for TLS.
+     * @param keyFile            Path to be key file to be used for TLS.
+     * @param tlsProtocolVersion The version of the TLS protocol
+     * @param healthServiceManager The healService to register new health contributors related to the listeners.
+     * @param type               The type of listener.
```

Comment:
Who sets this type? Is it someone external? Or is it just the derived classes?

If the latter, then I suggest defining an abstract method that returns the type and each implementation will return its appropriate type.

Reply:
Or better yet, just use `this.getClass().getSimpleName()`. This will give you the name of the class you are instantiating (not the base class, but the derived class). so it should achieve the same purpose.